### DISCUSSION TITLE
Error After updated today.

### DISCUSSION THREAD

**Discussion by KisakuKK77** [bmaltais/kohya_ss#920] [Q&A]
![螢幕擷取畫面 2023-06-05 153335](https://github.com/bmaltais/kohya_ss/assets/135601718/1d61220e-8a8c-45a5-8d13-763b92e911f5)

=============================================================
Modules installed outside the virtual environment were found.
This can cause issues. Please review the installed modules.

You can uninstall all local modules with:

deactivate
pip freeze > uninstall.txt
pip uninstall -y -r uninstall.txt

=============================================================

And Output Lora:
17:39:39-085515 INFO     nVidia toolkit detected
17:39:41-513735 INFO     Torch 2.0.1+cu118
17:39:41-529682 INFO     Torch backend: nVidia CUDA 11.8 cuDNN 8700
17:39:41-531676 INFO     Torch detected GPU: NVIDIA GeForce RTX 3080 VRAM 12288 Arch (8, 6) Cores 70
17:39:41-532672 INFO     Verifying requirements
17:39:43-560372 INFO     headless: False
17:39:43-562911 INFO     Load CSS...
Running on local URL:  http://127.0.0.1:7861

To create a public link, set `share=True` in `launch()`.
17:39:55-644836 INFO     Loading config...
17:39:55-693673 INFO     LoRA type changed...
17:40:41-622154 INFO     Start training LoRA Kohya LoCon ...
17:40:41-627137 INFO     Folder 100_EAV P: 46 images found
17:40:41-628134 INFO     Folder 100_EAV P: 4600 steps
17:40:41-629130 INFO     Total steps: 4600
17:40:41-630126 INFO     Train batch size: 2
17:40:41-632120 INFO     Gradient accumulation steps: 1.0
17:40:41-633117 INFO     Epoch: 1
17:40:41-634113 INFO     Regulatization factor: 1
17:40:41-635110 INFO     max_train_steps (4600 / 2 / 1.0 * 1 * 1) = 2300
17:40:41-636107 INFO     stop_text_encoder_training = 0
17:40:41-637103 INFO     lr_warmup_steps = 230
17:40:41-639097 INFO     accelerate launch --num_cpu_threads_per_process=2 "train_network.py"
                         --pretrained_model_name_or_path="E:\Game
                         2023\AIWAIFU\SD15\stable-diffusion-webui\models\Stable-diffusion\yesmix_v16Original.safetensors                         " --train_data_dir="C:\Test\Aya_Kujyou_Final_Gear\2" --resolution=512,512
                         --output_dir="C:\Test\Aya_Kujyou_Final_Gear\2" --network_alpha="128"
                         --save_model_as=safetensors --network_module=networks.lora --network_args conv_dim="1"
                         conv_alpha="1" --text_encoder_lr=5e-05 --unet_lr=0.0001 --network_dim=16
                         --output_name="Aya_Kujyou_Final_Gear-KK77-V1" --lr_scheduler_num_cycles="1"
                         --learning_rate="0.0001" --lr_scheduler="cosine" --lr_warmup_steps="230" --train_batch_size="2"                         --max_train_steps="2300" --save_every_n_epochs="1" --mixed_precision="fp16"
                         --save_precision="fp16" --seed="1" --caption_extension=".txt" --cache_latents
                         --optimizer_type="AdamW8bit" --max_data_loader_n_workers="1" --clip_skip=2
                         --bucket_reso_steps=64 --xformers --bucket_no_upscale --wandb_api_key="False"
Traceback (most recent call last):
  File "C:\Users\admin\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\admin\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\venv\Scripts\accelerate.exe\__main__.py", line 4, in <module>
  File "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\venv\lib\site-packages\accelerate\__init__.py", line 3, in <module>
    from .accelerator import Accelerator
  File "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\venv\lib\site-packages\accelerate\accelerator.py", line 39, in <module>
    from .tracking import LOGGER_TYPE_TO_CLASS, GeneralTracker, filter_trackers
  File "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\venv\lib\site-packages\accelerate\tracking.py", line 42, in <module>
    from torch.utils import tensorboard
  File "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\venv\lib\site-packages\torch\utils\tensorboard\__init__.py", line 12, in <module>
    from .writer import FileWriter, SummaryWriter  # noqa: F401
  File "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\venv\lib\site-packages\torch\utils\tensorboard\writer.py", line 16, in <module>
    from ._embedding import (
  File "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\venv\lib\site-packages\torch\utils\tensorboard\_embedding.py", line 9, in <module>
    _HAS_GFILE_JOIN = hasattr(tf.io.gfile, "join")
  File "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\venv\lib\site-packages\tensorboard\lazy.py", line 65, in __getattr__
    return getattr(load_once(self), attr_name)
AttributeError: module 'tensorflow' has no attribute 'io'

**Comment by KisakuKK77** [comment]
Now, Kohya ss is really bad. It is too slow to make lora.

Before 2023-06-03 I still can use Torch 2 and cudnn, It so fast, 

But now It's dead...So sad

**Comment by KisakuKK77** [comment]
Now, what is this?

=============================================================
Modules installed outside the virtual environment were found.
This can cause issues. Please review the installed modules.

You can uninstall all local modules with:

deactivate
pip freeze > uninstall.txt
pip uninstall -y -r uninstall.txt

=============================================================

12:14:18-662189 INFO     nVidia toolkit detected
12:14:19-685163 INFO     Torch 1.12.1+cu116
12:14:19-702673 INFO     Torch backend: nVidia CUDA 11.6 cuDNN 8901
12:14:19-704670 INFO     Torch detected GPU: NVIDIA GeForce RTX 3080 VRAM 12288 Arch (8, 6) Cores 70
12:14:19-705672 INFO     Verifying requirements
12:14:19-751183 WARNING  Package wrong version: huggingface-hub 0.15.1 required 0.13.3
12:14:19-752182 INFO     Installing package: huggingface-hub==0.13.3
12:14:26-545070 INFO     headless: False
12:14:26-549068 INFO     Load CSS...
Running on local URL:  http://127.0.0.1:7861

To create a public link, set `share=True` in `launch()`.
12:14:50-182032 INFO     Loading config...
12:14:50-261548 INFO     LoRA type changed...
12:14:59-552952 INFO     Start training LoRA Kohya LoCon ...
12:14:59-555950 INFO     Folder 50_EAV P: 46 images found
12:14:59-556952 INFO     Folder 50_EAV P: 2300 steps
12:14:59-557950 INFO     Total steps: 2300
12:14:59-558950 INFO     Train batch size: 2
12:14:59-559950 INFO     Gradient accumulation steps: 1.0
12:14:59-560950 INFO     Epoch: 1
12:14:59-561950 INFO     Regulatization factor: 1
12:14:59-562949 INFO     max_train_steps (2300 / 2 / 1.0 * 1 * 1) = 1150
12:14:59-564952 INFO     stop_text_encoder_training = 0
12:14:59-565951 INFO     lr_warmup_steps = 115
12:14:59-567952 INFO     accelerate launch --num_cpu_threads_per_process=2 "train_network.py"
                         --pretrained_model_name_or_path="E:\Game
                         2023\AIWAIFU\SD15\stable-diffusion-webui\models\Stable-diffusion\yesmix_v16Original.safetensors
                         " --train_data_dir="E:\Game
                         2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\YMAKELORA\Model\20230605\Aya_Kujyou_Final_Gear\
                         Aya_Kujyou_Final_Gear" --resolution=512,512 --output_dir="E:\Game
                         2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\YMAKELORA\Model\20230605\Aya_Kujyou_Final_Gear\
                         Aya_Kujyou_Final_Gear" --network_alpha="128" --save_model_as=safetensors
                         --network_module=networks.lora --network_args conv_dim="1" conv_alpha="1"
                         --text_encoder_lr=5e-05 --unet_lr=0.0001 --network_dim=16
                         --output_name="Aya_Kujyou_Final_Gear-KK77-V3" --lr_scheduler_num_cycles="1"
                         --learning_rate="0.0001" --lr_scheduler="cosine" --lr_warmup_steps="115" --train_batch_size="2"
                         --max_train_steps="1150" --save_every_n_epochs="1" --mixed_precision="fp16"
                         --save_precision="fp16" --seed="1" --caption_extension=".txt" --cache_latents
                         --optimizer_type="AdamW8bit" --max_data_loader_n_workers="1" --clip_skip=2
                         --bucket_reso_steps=64 --xformers --bucket_no_upscale --wandb_api_key="False"
prepare tokenizer
Using DreamBooth method.
ignore directory without repeats / 繰り返し回数のないディレクトリを無視します: sample
prepare images.
found directory E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\YMAKELORA\Model\20230605\Aya_Kujyou_Final_Gear\Aya_Kujyou_Final_Gear\50_EAV P contains 46 image files
2300 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 2
  resolution: (512, 512)
  enable_bucket: False

  [Subset 0 of Dataset 0]
    image_dir: "E:\Game 2023\AIWAIFU\kohya_ss20230512\kohya_ss\kohya_ss\YMAKELORA\Model\20230605\Aya_Kujyou_Final_Gear\Aya_Kujyou_Final_Gear\50_EAV P"
    image_count: 46
    num_repeats: 50
    shuffle_caption: False
    keep_tokens: 0
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epoches: 0
    caption_tag_dropout_rate: 0.0
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    token_warmup_min: 1,
    token_warmup_step: 0,
    is_reg: False
    class_tokens: EAV P
    caption_extension: .txt


[Dataset 0]
loading image sizes.
100%|████████████████████████████████████████████████████████████████████████████████| 46/46 [00:00<00:00, 5749.39it/s]
prepare dataset
preparing accelerator
Using accelerator 0.15.0 or above.
loading model for process 0/1
load StableDiffusion checkpoint: E:\Game 2023\AIWAIFU\SD15\stable-diffusion-webui\models\Stable-diffusion\yesmix_v16Original.safetensors
loading u-net: <All keys matched successfully>
loading vae: <All keys matched successfully>
loading text encoder: <All keys matched successfully>
CrossAttention.forward has been replaced to enable xformers.
import network module: networks.lora
[Dataset 0]
caching latents.
  0%|                                                                                           | 0/46 [00:00<?, ?it/s]
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ E:\Game 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\train_network.py:864 in <module>                │
│                                                                                                  │
│   861 │   args = parser.parse_args()                                                             │
│   862 │   args = train_util.read_config_from_file(args, parser)                                  │
│   863 │                                                                                          │
│ ❱ 864 │   train(args)                                                                            │
│   865                                                                                            │
│                                                                                                  │
│ E:\Game 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\train_network.py:194 in train                   │
│                                                                                                  │
│   191 │   │   vae.requires_grad_(False)                                                          │
│   192 │   │   vae.eval()                                                                         │
│   193 │   │   with torch.no_grad():                                                              │
│ ❱ 194 │   │   │   train_dataset_group.cache_latents(vae, args.vae_batch_size, args.cache_laten   │
│   195 │   │   vae.to("cpu")                                                                      │
│   196 │   │   if torch.cuda.is_available():                                                      │
│   197 │   │   │   torch.cuda.empty_cache()                                                       │
│                                                                                                  │
│ E:\Game 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\library\train_util.py:1422 in cache_latents     │
│                                                                                                  │
│   1419 │   def cache_latents(self, vae, vae_batch_size=1, cache_to_disk=False, is_main_process=  │
│   1420 │   │   for i, dataset in enumerate(self.datasets):                                       │
│   1421 │   │   │   print(f"[Dataset {i}]")                                                       │
│ ❱ 1422 │   │   │   dataset.cache_latents(vae, vae_batch_size, cache_to_disk, is_main_process)    │
│   1423 │                                                                                         │
│   1424 │   def is_latent_cacheable(self) -> bool:                                                │
│   1425 │   │   return all([dataset.is_latent_cacheable() for dataset in self.datasets])          │
│                                                                                                  │
│ E:\Game 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\library\train_util.py:814 in cache_latents      │
│                                                                                                  │
│    811 │   │   │   img_tensors = torch.stack(images, dim=0)                                      │
│    812 │   │   │   img_tensors = img_tensors.to(device=vae.device, dtype=vae.dtype)              │
│    813 │   │   │                                                                                 │
│ ❱  814 │   │   │   latents = vae.encode(img_tensors).latent_dist.sample().to("cpu")              │
│    815 │   │   │                                                                                 │
│    816 │   │   │   for info, latent in zip(batch, latents):                                      │
│    817 │   │   │   │   if cache_to_disk:                                                         │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\diffusers\models\vae.py:566 in   │
│ encode                                                                                           │
│                                                                                                  │
│   563 │   │   self.use_slicing = False                                                           │
│   564 │                                                                                          │
│   565 │   def encode(self, x: torch.FloatTensor, return_dict: bool = True) -> AutoencoderKLOut   │
│ ❱ 566 │   │   h = self.encoder(x)                                                                │
│   567 │   │   moments = self.quant_conv(h)                                                       │
│   568 │   │   posterior = DiagonalGaussianDistribution(moments)                                  │
│   569                                                                                            │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\torch\nn\modules\module.py:1130  │
│ in _call_impl                                                                                    │
│                                                                                                  │
│   1127 │   │   # this function, and just call forward.                                           │
│   1128 │   │   if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks o  │
│   1129 │   │   │   │   or _global_forward_hooks or _global_forward_pre_hooks):                   │
│ ❱ 1130 │   │   │   return forward_call(*input, **kwargs)                                         │
│   1131 │   │   # Do not call functions when jit is used                                          │
│   1132 │   │   full_backward_hooks, non_full_backward_hooks = [], []                             │
│   1133 │   │   if self._backward_hooks or _global_backward_hooks:                                │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\diffusers\models\vae.py:130 in   │
│ forward                                                                                          │
│                                                                                                  │
│   127 │                                                                                          │
│   128 │   def forward(self, x):                                                                  │
│   129 │   │   sample = x                                                                         │
│ ❱ 130 │   │   sample = self.conv_in(sample)                                                      │
│   131 │   │                                                                                      │
│   132 │   │   # down                                                                             │
│   133 │   │   for down_block in self.down_blocks:                                                │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\torch\nn\modules\module.py:1130  │
│ in _call_impl                                                                                    │
│                                                                                                  │
│   1127 │   │   # this function, and just call forward.                                           │
│   1128 │   │   if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks o  │
│   1129 │   │   │   │   or _global_forward_hooks or _global_forward_pre_hooks):                   │
│ ❱ 1130 │   │   │   return forward_call(*input, **kwargs)                                         │
│   1131 │   │   # Do not call functions when jit is used                                          │
│   1132 │   │   full_backward_hooks, non_full_backward_hooks = [], []                             │
│   1133 │   │   if self._backward_hooks or _global_backward_hooks:                                │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\torch\nn\modules\conv.py:457 in  │
│ forward                                                                                          │
│                                                                                                  │
│    454 │   │   │   │   │   │   self.padding, self.dilation, self.groups)                         │
│    455 │                                                                                         │
│    456 │   def forward(self, input: Tensor) -> Tensor:                                           │
│ ❱  457 │   │   return self._conv_forward(input, self.weight, self.bias)                          │
│    458                                                                                           │
│    459 class Conv3d(_ConvNd):                                                                    │
│    460 │   __doc__ = r"""Applies a 3D convolution over an input signal composed of several inpu  │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\torch\nn\modules\conv.py:453 in  │
│ _conv_forward                                                                                    │
│                                                                                                  │
│    450 │   │   │   return F.conv2d(F.pad(input, self._reversed_padding_repeated_twice, mode=sel  │
│    451 │   │   │   │   │   │   │   weight, bias, self.stride,                                    │
│    452 │   │   │   │   │   │   │   _pair(0), self.dilation, self.groups)                         │
│ ❱  453 │   │   return F.conv2d(input, weight, bias, self.stride,                                 │
│    454 │   │   │   │   │   │   self.padding, self.dilation, self.groups)                         │
│    455 │                                                                                         │
│    456 │   def forward(self, input: Tensor) -> Tensor:                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
RuntimeError: "slow_conv2d_cpu" not implemented for 'Half'
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Users\admin\AppData\Local\Programs\Python\Python310\lib\runpy.py:196 in _run_module_as_main   │
│                                                                                                  │
│   193 │   main_globals = sys.modules["__main__"].__dict__                                        │
│   194 │   if alter_argv:                                                                         │
│   195 │   │   sys.argv[0] = mod_spec.origin                                                      │
│ ❱ 196 │   return _run_code(code, main_globals, None,                                             │
│   197 │   │   │   │   │    "__main__", mod_spec)                                                 │
│   198                                                                                            │
│   199 def run_module(mod_name, init_globals=None,                                                │
│                                                                                                  │
│ C:\Users\admin\AppData\Local\Programs\Python\Python310\lib\runpy.py:86 in _run_code              │
│                                                                                                  │
│    83 │   │   │   │   │      __loader__ = loader,                                                │
│    84 │   │   │   │   │      __package__ = pkg_name,                                             │
│    85 │   │   │   │   │      __spec__ = mod_spec)                                                │
│ ❱  86 │   exec(code, run_globals)                                                                │
│    87 │   return run_globals                                                                     │
│    88                                                                                            │
│    89 def _run_module_code(code, init_globals=None,                                              │
│                                                                                                  │
│ in <module>:7                                                                                    │
│                                                                                                  │
│   4 from accelerate.commands.accelerate_cli import main                                          │
│   5 if __name__ == '__main__':                                                                   │
│   6 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 7 │   sys.exit(main())                                                                         │
│   8                                                                                              │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\accelerate\commands\accelerate_c │
│ li.py:45 in main                                                                                 │
│                                                                                                  │
│   42 │   │   exit(1)                                                                             │
│   43 │                                                                                           │
│   44 │   # Run                                                                                   │
│ ❱ 45 │   args.func(args)                                                                         │
│   46                                                                                             │
│   47                                                                                             │
│   48 if __name__ == "__main__":                                                                  │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py:11 │
│ 04 in launch_command                                                                             │
│                                                                                                  │
│   1101 │   elif defaults is not None and defaults.compute_environment == ComputeEnvironment.AMA  │
│   1102 │   │   sagemaker_launcher(defaults, args)                                                │
│   1103 │   else:                                                                                 │
│ ❱ 1104 │   │   simple_launcher(args)                                                             │
│   1105                                                                                           │
│   1106                                                                                           │
│   1107 def main():                                                                               │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230606_1\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py:56 │
│ 7 in simple_launcher                                                                             │
│                                                                                                  │
│    564 │   process = subprocess.Popen(cmd, env=current_env)                                      │
│    565 │   process.wait()                                                                        │
│    566 │   if process.returncode != 0:                                                           │
│ ❱  567 │   │   raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)       │
│    568                                                                                           │
│    569                                                                                           │
│    570 def multi_gpu_launcher(args):                                                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
CalledProcessError: Command '['E:\\Game 2023\\AIWAIFU\\kohya_ss20230606_1\\kohya_ss\\venv\\Scripts\\python.exe',
'train_network.py', '--pretrained_model_name_or_path=E:\\Game
2023\\AIWAIFU\\SD15\\stable-diffusion-webui\\models\\Stable-diffusion\\yesmix_v16Original.safetensors',
'--train_data_dir=E:\\Game
2023\\AIWAIFU\\kohya_ss20230512\\kohya_ss\\kohya_ss\\YMAKELORA\\Model\\20230605\\Aya_Kujyou_Final_Gear\\Aya_Kujyou_Final
_Gear', '--resolution=512,512', '--output_dir=E:\\Game
2023\\AIWAIFU\\kohya_ss20230512\\kohya_ss\\kohya_ss\\YMAKELORA\\Model\\20230605\\Aya_Kujyou_Final_Gear\\Aya_Kujyou_Final
_Gear', '--network_alpha=128', '--save_model_as=safetensors', '--network_module=networks.lora', '--network_args',
'conv_dim=1', 'conv_alpha=1', '--text_encoder_lr=5e-05', '--unet_lr=0.0001', '--network_dim=16',
'--output_name=Aya_Kujyou_Final_Gear-KK77-V3', '--lr_scheduler_num_cycles=1', '--learning_rate=0.0001',
'--lr_scheduler=cosine', '--lr_warmup_steps=115', '--train_batch_size=2', '--max_train_steps=1150',
'--save_every_n_epochs=1', '--mixed_precision=fp16', '--save_precision=fp16', '--seed=1', '--caption_extension=.txt',
'--cache_latents', '--optimizer_type=AdamW8bit', '--max_data_loader_n_workers=1', '--clip_skip=2',
'--bucket_reso_steps=64', '--xformers', '--bucket_no_upscale', '--wandb_api_key=False']' returned non-zero exit status
1.

**Comment by KisakuKK77** [comment]
05:01:52-077763 INFO     Stopping tensorboard process...
Traceback (most recent call last):
  File "E:\Game 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\gradio\routes.py", line 422, in run_predict
    output = await app.get_blocks().process_api(
  File "E:\Game 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\gradio\blocks.py", line 1323, in process_api
    result = await self.call_function(
  File "E:\Game 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\gradio\blocks.py", line 1051, in call_function
    prediction = await anyio.to_thread.run_sync(
  File "E:\Game 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "E:\Game 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "E:\Game 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "E:\Game 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\library\tensorboard_gui.py", line 49, in stop_tensorboard
    tensorboard_proc.kill()
AttributeError: 'NoneType' object has no attribute 'kill'

**Comment by KisakuKK77** [comment]
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Users\admin\AppData\Local\Programs\Python\Python310\lib\runpy.py:196 in _run_module_as_main   │
│                                                                                                  │
│   193 │   main_globals = sys.modules["__main__"].__dict__                                        │
│   194 │   if alter_argv:                                                                         │
│   195 │   │   sys.argv[0] = mod_spec.origin                                                      │
│ ❱ 196 │   return _run_code(code, main_globals, None,                                             │
│   197 │   │   │   │   │    "__main__", mod_spec)                                                 │
│   198                                                                                            │
│   199 def run_module(mod_name, init_globals=None,                                                │
│                                                                                                  │
│ C:\Users\admin\AppData\Local\Programs\Python\Python310\lib\runpy.py:86 in _run_code              │
│                                                                                                  │
│    83 │   │   │   │   │      __loader__ = loader,                                                │
│    84 │   │   │   │   │      __package__ = pkg_name,                                             │
│    85 │   │   │   │   │      __spec__ = mod_spec)                                                │
│ ❱  86 │   exec(code, run_globals)                                                                │
│    87 │   return run_globals                                                                     │
│    88                                                                                            │
│    89 def _run_module_code(code, init_globals=None,                                              │
│                                                                                                  │
│ in <module>:7                                                                                    │
│                                                                                                  │
│   4 from accelerate.commands.accelerate_cli import main                                          │
│   5 if __name__ == '__main__':                                                                   │
│   6 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 7 │   sys.exit(main())                                                                         │
│   8                                                                                              │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\accelerate\commands\accelerate_c │
│ li.py:45 in main                                                                                 │
│                                                                                                  │
│   42 │   │   exit(1)                                                                             │
│   43 │                                                                                           │
│   44 │   # Run                                                                                   │
│ ❱ 45 │   args.func(args)                                                                         │
│   46                                                                                             │
│   47                                                                                             │
│   48 if __name__ == "__main__":                                                                  │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py:10 │
│ 00 in launch_command                                                                             │
│                                                                                                  │
│    997 │   warned = []                                                                           │
│    998 │   # Get the default from the config file.                                               │
│    999 │   if args.config_file is not None or os.path.isfile(default_config_file) and not args.  │
│ ❱ 1000 │   │   defaults = load_config_from_file(args.config_file)                                │
│   1001 │   │   if (                                                                              │
│   1002 │   │   │   not args.multi_gpu                                                            │
│   1003 │   │   │   and not args.tpu                                                              │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\accelerate\commands\config\confi │
│ g_args.py:64 in load_config_from_file                                                            │
│                                                                                                  │
│    61 │   │   │   │   config_class = ClusterConfig                                               │
│    62 │   │   │   else:                                                                          │
│    63 │   │   │   │   config_class = SageMakerConfig                                             │
│ ❱  64 │   │   │   return config_class.from_yaml_file(yaml_file=config_file)                      │
│    65                                                                                            │
│    66                                                                                            │
│    67 @dataclass                                                                                 │
│                                                                                                  │
│ E:\Game                                                                                          │
│ 2023\AIWAIFU\kohya_ss20230607_1\kohya_ss\venv\lib\site-packages\accelerate\commands\config\confi │
│ g_args.py:122 in from_yaml_file                                                                  │
│                                                                                                  │
│   119 │   │   if "dynamo_backend" not in config_dict:                                            │
│   120 │   │   │   config_dict["dynamo_backend"] = DynamoBackend.NO                               │
│   121 │   │                                                                                      │
│ ❱ 122 │   │   return cls(**config_dict)                                                          │
│   123 │                                                                                          │
│   124 │   def to_yaml_file(self, yaml_file):                                                     │
│   125 │   │   with open(yaml_file, "w", encoding="utf-8") as f:                                  │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
TypeError: ClusterConfig.__init__() got an unexpected keyword argument 'tpu_env'

**Comment by KisakuKK77** [comment]
2023-06-07T18:47:51.630ZE [8784:ShellIpcClient] shell_ipc_client.cc:138:Connect Can't connect to socket at: \\.\Pipe\GoogleDriveFSPipe_admin_shell
2023-06-07T18:47:51.633ZE [8784:ShellIpcClient] shell_ipc_client.cc:621:operator() Failed to connect to the server: NOT_FOUND: Can't connect to socket at: \\.\Pipe\GoogleDriveFSPipe_admin_shell [type.googleapis.com/drive.ds.Status='UNAVAILABLE_RESOURCE']
=== Source Location Trace: ===
apps/drive/fs/ipc/shell_ipc_client.cc:139

2023-06-07T18:47:51.633ZE [13980:ShellIpcClient] shell_ipc_client.cc:138:Connect Can't connect to socket at: \\.\Pipe\GoogleDriveFSPipe_admin_shell
2023-06-07T18:47:51.634ZE [13980:ShellIpcClient] shell_ipc_client.cc:621:operator() Failed to connect to the server: NOT_FOUND: Can't connect to socket at: \\.\Pipe\GoogleDriveFSPipe_admin_shell [type.googleapis.com/drive.ds.Status='UNAVAILABLE_RESOURCE']
=== Source Location Trace: ===
apps/drive/fs/ipc/shell_ipc_client.cc:139

2023-06-07T18:47:51.634ZE [8944:ShellIpcClient] shell_ipc_client.cc:138:Connect Can't connect to socket at: \\.\Pipe\GoogleDriveFSPipe_admin_shell
2023-06-07T18:47:51.634ZE [8944:ShellIpcClient] shell_ipc_client.cc:621:operator() Failed to connect to the server: NOT_FOUND: Can't connect to socket at: \\.\Pipe\GoogleDriveFSPipe_admin_shell [type.googleapis.com/drive.ds.Status='UNAVAILABLE_RESOURCE']
=== Source Location Trace: ===
apps/drive/fs/ipc/shell_ipc_client.cc:139

**Comment by Omarkh12** [comment]
Im having the same error trying to install it

**Comment by KisakuKK77** [comment]
Before 2023-06-03 I still can use Torch 2 and cudnn, It so fast,

But now using Torch 1 It is extremely slow....

**Comment by KisakuKK77** [comment]
v21.7.6
=============================================================
Modules installed outside the virtual environment were found.
This can cause issues. Please review the installed modules.

You can uninstall all local modules with:

deactivate
pip freeze > uninstall.txt
pip uninstall -y -r uninstall.txt

=============================================================

A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'

Speed also extremely slow....

**Comment by dougvan1701** [comment]
I had this problem after running upgrade.bat yesterday, I simply deleted my kohya_ss folder and re-installed. Worked just fine. Only issue I have is it complaining about Triton missing but that hasn't affected building my models. Lora seems to work just fine and not super slow for me.

**Comment by mistermcluvin** [comment]
Yeah, I made the mistake of upgrading my old version of it yesterday and it just won't work. I got similar errors. I finally was able to get it to create a Lora (with the Triton warning), then I got... Failed to match keys when loading Lora in A1111. Argh. I wish I just stayed on the old version. Oh well, what can you do. It's not like it's paid software and I do appreciate the creators and developers. It's not like I can create these things myself.

**Comment by Lolamombobdoda** [comment]
I got similar issues after updating once and it was because it was trying to update into the wrong folder.
I don't remember exactly what it was but I think it was putting stuff directly into the kohya_ss folder instead of kohya_ss\venv, but don't quote me on that.

Reinstalling fixed it for me.
Reinstalling have fixed most issue for me, except for xformers not working but that got fixed in an update a couple of months ago!

**Comment by Boldor83** [comment]
i still get this every install i try.
following instructions:
https://github.com/bmaltais/kohya_ss under windows.
Then trying 
deactivate
pip freeze > uninstall.txt
pip uninstall -y -r uninstall.txt
still gives this message and also when i try pip install -r requirements.txt afterwards.
